frontend/src/components/button/button.tsx

```diff
@@ -0,0 +1,68 @@
+import { type ReactNode } from 'react';
+
+type ButtonVariant = 'primary' | 'secondary' | 'tertiary';
+type ButtonSize = 'small' | 'medium';
+type ButtonType = 'button' | 'submit';
+
+type ButtonProperties = {
+    children: ReactNode;
+    size: ButtonSize;
+    variant: ButtonVariant;
+    type?: ButtonType;
+    disabled?: boolean;
```

Comment:
```suggestion
    isDisabled?: boolean;
```
https://github.com/BinaryStudioAcademy/quality-criteria/blob/production/src/javascript.md - B19 rule here